src/kudu/consensus/raft_consensus.cc

```diff
@@ -2678,6 +2684,19 @@ void RaftConsensus::DoElectionCallback(ElectionReason reason, const ElectionResu
   }
 }
 
+void RaftConsensus::NestedElectionDecisionCallback(
+    ElectionReason reason, const ElectionResult& result) {
+  DoElectionCallback(reason, result);
+  const bool was_pre_election = result.vote_request.is_pre_election();
+  UniqueLock lock(lock_);
+  const bool runClientElectionDecisionCallback =
+      (cmeta_->active_role() == RaftPeerPB::LEADER) || !was_pre_election;
+  lock.unlock();
+  if (runClientElectionDecisionCallback) {
+    edcb_(result);
```

Comment:
edcb_ can be changing when RegisterElectionOutcomeCallback is called,
which is unlikely. Ideally you should not allow this callback to every change,
and that can be ensured by setting it before Start.

Reply:
agree